proto/api.proto

```diff
@@ -240,6 +279,15 @@ message ScanResponse {
   optional ResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
   // Empty if no rows were scanned.
   repeated KeyValue rows = 2 [(gogoproto.nullable) = false];
+  // If the scan is for a table scan the following fields are populated.
+  //
+  // Columns in the table scan.
+  repeated bytes column_names = 3;
+  message Row {
+    // Values for the different columns specified above.
+    repeated Value value = 1;
+  }
+  repeated Row table_rows = 4;
```

Comment:
Ah, I see you've kept the names and values separate here. I'd do this for all of the responses.